bundles/org.openhab.binding.pulseaudio/src/main/java/org/openhab/binding/pulseaudio/internal/PulseAudioAudioSink.java

```diff
@@ -0,0 +1,188 @@
+import javazoom.spi.mpeg.sampled.file.MpegAudioFileReader;
+
+import javax.sound.sampled.AudioInputStream;
+import javax.sound.sampled.UnsupportedAudioFileException;
+
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.pulseaudio.internal.handler.PulseaudioHandler;
+import org.openhab.core.audio.AudioFormat;
+import org.openhab.core.audio.AudioSink;
+import org.openhab.core.audio.AudioStream;
+import org.openhab.core.audio.FixedLengthAudioStream;
+import org.openhab.core.audio.UnsupportedAudioFormatException;
+import org.openhab.core.audio.UnsupportedAudioStreamException;
+import org.openhab.core.library.types.PercentType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The audio sink for openhab, implemented by a connection to a pulseaudio sink
+ *
+ * @author Gwendal Roulleau - Initial contribution
+ *
+ */
+public class PulseAudioAudioSink implements AudioSink {
+
+    private final Logger logger = LoggerFactory.getLogger(PulseAudioAudioSink.class);
+
+    private static final HashSet<AudioFormat> SUPPORTED_FORMATS = new HashSet<>();
+    private static final HashSet<Class<? extends AudioStream>> SUPPORTED_STREAMS = new HashSet<>();
+
+    private PulseaudioHandler pulseaudioHandler;
+
+    private Socket clientSocket;
+
+    static {
+        SUPPORTED_FORMATS.add(AudioFormat.WAV);
+        SUPPORTED_FORMATS.add(AudioFormat.MP3);
+        SUPPORTED_STREAMS.add(FixedLengthAudioStream.class);
+    }
+
+    public PulseAudioAudioSink(PulseaudioHandler pulseaudioHandler) {
+        this.pulseaudioHandler = pulseaudioHandler;
+    }
+
+    @Override
+    public String getId() {
+        return pulseaudioHandler.getThing().getUID().toString();
+    }
+
+    @Override
+    public @Nullable String getLabel(@Nullable Locale locale) {
+        return pulseaudioHandler.getThing().getLabel();
+    }
+
+    /**
+     * Convert MP3 to PCM, as this is the only possible format
+     *
+     * @param input
+     * @return
+     */
+    private InputStream getPCMStreamFromMp3Stream(InputStream input) {
+        try {
+            MpegAudioFileReader mpegAudioFileReader = new MpegAudioFileReader();
+            AudioInputStream sourceAIS = mpegAudioFileReader.getAudioInputStream(input);
+            javax.sound.sampled.AudioFormat sourceFormat = sourceAIS.getFormat();
+
+            MpegFormatConversionProvider mpegconverter = new MpegFormatConversionProvider();
+            javax.sound.sampled.AudioFormat convertFormat = new javax.sound.sampled.AudioFormat(
+                    javax.sound.sampled.AudioFormat.Encoding.PCM_SIGNED, sourceFormat.getSampleRate(), 16,
+                    sourceFormat.getChannels(), sourceFormat.getChannels() * 2, sourceFormat.getSampleRate(), false);
+
+            return mpegconverter.getAudioInputStream(convertFormat, sourceAIS);
+
+        } catch (IOException | UnsupportedAudioFileException e) {
+            logger.error("Cannot convert this mp3 stream to pcm stream", e);
```

Comment:
Bindings should only log to error if something severe happened, like the detection of a bug in your code. This could be warn. See this link for a description of the log levels: https://www.openhab.org/docs/developer/guidelines.html#f-logging

It doesn't seem reasonable to log the stacktrace when the network fails:
```suggestion
            logger.error("Cannot convert this mp3 stream to pcm stream: {}", e.getMessage());
```

Please check all.